src/emulate.c

```diff
@@ -665,30 +666,57 @@ static inline bool op_jal(struct riscv_t *rv, uint32_t insn)
     return false;
 }
 
+static int64_t time_us(void)
+{
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    uint64_t value = (uint64_t) tv.tv_sec * 1000000; /* uint64_t - seconds */
+    value += (uint32_t) tv.tv_usec;                  /* microseconds */
+    return value;
+};
+
 /* get a pointer to a CSR */
 static uint32_t *csr_get_ptr(struct riscv_t *rv, uint32_t csr)
 {
     switch (csr) {
-    case CSR_CYCLE:
-        return (uint32_t *) (&rv->csr_cycle) + 0;
-    case CSR_CYCLEH:
-        return (uint32_t *) (&rv->csr_cycle) + 1;
-    case CSR_MSTATUS:
+    // Machine Trap Setup
```

Comment:
Always use C-style comments. i.e., use `/* comments */` instead of the inlined `//`. See file `CONTRIBUTING.md ` in top-level directory for details.